### ISSUE TITLE
[ADD] Option to have test in the same directory as src or in different location

### ISSUE THREAD

**Issue by Abdizriel** [ericmdantas/generator-ng-fullstack#173]
Right now src files and test files are separated in different folders for client, server and tests.

```
client/
server/
tests/client/
tests/server/
tests/e2e/
```

We should allow user to choose if wants to have it in separate directories or test and src in the same
In case when user would like to have in separate directories it would stay like it is now but in case of having them in one directory it would be done like

Angular 1:
```
client/dev/{feature}/controllers/{name}.js
client/dev/{feature}/controllers/{name}.spec.js
client/dev/{feature}/directives/{name}.js
client/dev/{feature}/directives/{name}.spec.js
client/dev/{feature}/services/{name}.js
client/dev/{feature}/services/{name}.spec.js
client/dev/{feature}/decorators/{name}.js
client/dev/{feature}/decorators/{name}.spec.js
client/dev/{feature}/resource/{name}.js
client/dev/{feature}/resource/{name}.spec.js
client/dev/{feature}/models/{name}.js
client/dev/{feature}/models/{name}.spec.js
client/dev/{feature}/filters/{name}.js
client/dev/{feature}/filters/{name}.spec.js
client/dev/{feature}/factory/{name}.js
client/dev/{feature}/factory/{name}.spec.js
client/dev/{feature}/templates/{name}.html
client/dev/{feature}/styles/{name}.css
```

Angular 2
```
client/dev/{feature}/components/{name}.ts
client/dev/{feature}/components/{name}.spec.ts
client/dev/{feature}/directives/{name}.ts
client/dev/{feature}/directives/{name}.spec.ts
client/dev/{feature}/services/{name}.ts
client/dev/{feature}/services/{name}.spec.ts
client/dev/{feature}/factory/{name}.ts
client/dev/{feature}/factory/{name}.spec.ts
client/dev/{feature}/models/{name}.ts
client/dev/{feature}/models/{name}.spec.ts
client/dev/{feature}/pipes/{name}.ts
client/dev/{feature}/pipes/{name}.spec.ts
client/dev/{feature}/templates/{name}.html
client/dev/{feature}/styles/{name}.css
```

Node Server
```
server/api/{feature}/controllers/{name}.js
server/api/{feature}/controllers/{name}.spec.js
server/api/{feature}/dao/{name}.js
server/api/{feature}/dao/{name}.spec.js
server/api/{feature}/models/{name}.js
server/api/{feature}/models/{name}.spec.js
server/api/{feature}/routes/{name}.js
server/api/{feature}/routes/{name}.spec.js
```

Go Server
```
server/api/{feature}/controller/{name}.go
server/api/{feature}/controller/{name}_test.go
server/api/{feature}/dao/{name}.go
server/api/{feature}/dao/{name}_test.go
server/api/{feature}/model/{name}.go
server/api/{feature}/model/{name}_test.go
server/api/{feature}/routes/{name}.go
server/api/{feature}/routes/{name}_test.go
```

and separate directory in main directory for e2e tests

**Comment by ericmdantas** [comment]
Just to point out: There should be no option to have go's src and tests separated - they only work together (in the same folder);

**Comment by ericmdantas** [comment]
Also, is there a reason why people choose `.spec` over `_test`? I always liked the former better, :smile:

**Comment by Abdizriel** [comment]
I don't know reason but I think it's because of shortcut of specification.

**Comment by ericmdantas** [comment]
On a side note, when you're working on the changes, can you work on top of 1.9, please? So it'll be easier for us to merge.

**Comment by Abdizriel** [comment]
Sure, no problem.

**Comment by ericmdantas** [comment]
Implemented in #176.
